recon/state.mmd

```diff
@@ -0,0 +1,21 @@
+stateDiagram-v2
```

Comment:
Do these diagram svgs show up in any of the README.md docs? or the rust doc?
I don't see where they are used.